docs: What's new in `nwaku`?

jm-clius opened the pull request:
Adds an update post on "What's new in `nwaku`?"

The aim of this post is at least threefold:
1. Keep the community updated about recent developments in `nwaku`
2. Synchronise again on the current and future focus areas for `nwaku`
3. Make the name change from `nim-waku` to `nwaku` official - as soon as this post gets approved, I'll update main docs, the Discord channel and Github repo to reflect the new name. See https://github.com/status-im/nim-waku/issues/868

I'll appreciate any feedback. Also let me know of any major oversights.

oskarth commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+date:   2022-04-11 10:00:00 +0200
+author: hanno
+published: true
+permalink: /nwaku-update-2022-04
```
Maybe /introducing-nwaku

jm-clius replied:
Done in [05b607c](https://github.com/vacp2p/vac.dev/pull/55/commits/05b607c2812a885741e7f51bc0b465aff978a26e)

oskarth commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+## Background
+
+If you've been following our [research log](https://vac.dev/research-log/),
+you'll know that many things have happened in the world of Waku v2 since [our last general update](./2020-02-14-waku-update.md).
```
I would consider this our most general update probably https://vac.dev/waku-v2-ethereum-coscup

D4nte replied:
This link is incorrect, need to use the one present in the `permalink` metadata of the target post.

oskarth commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+The [Nim client](https://github.com/status-im/nim-waku/tree/d2fccb5220144893f994a67f2cc26661247f101f/waku/v2), is our reference implementation,
+developed by the research team in parallel with the specs
+and building on a home-grown implementation of [`libp2p`](https://github.com/status-im/nim-libp2p).
+We believe the Nim client is suitable to run as [a standalone adaptive node](./2020-07-01-waku-v2-pitch.md),
```
Stronger phrase, not "believe" but "is"

jm-clius replied:
Done in [05b607c](https://github.com/vacp2p/vac.dev/pull/55/commits/05b607c2812a885741e7f51bc0b465aff978a26e)

oskarth commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+`nwaku` nodes that persist historical messages on disk now manage their own storage size based on the `--store-capacity`.
+This can significantly improve node start-up times.
+
+Doubtless more issues impacting stability will be addressed in future as `nwaku` matures,
```
I find doubtless usage here a bit odd

oskarth commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+Doubtless more issues impacting stability will be addressed in future as `nwaku` matures,
+but we've noticed a marked improvement in the stability of running `nwaku` nodes.
+These include environments where `nwaku` nodes are expected to run with a long uptime,
+such as the Vac team's own fleets.
```
qualify as experimental bootstrapping fleet? to play down importance etc

oskarth commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+
+This is a DHT-based discovery mechanism adapted to store and relay _node records_.
+Our implementation is based on [Ethereum's Discovery v5 protocol](https://github.com/ethereum/devp2p/blob/fa6428ada7385c13551873b2ae6ad2457c228eb8/discv5/discv5-theory.md)
+with some [minor modifications](https://rfc.vac.dev/spec/33/) to isolate our discovery network from that of Ethereum.
```
Maybe link to forum thread / mentioning upcoming write-up here regarding decision choice?

oskarth commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+An early addition to our suite of protocols was [an extension of `11/WAKU-RELAY`](https://rfc.vac.dev/spec/32/)
+that provided spam protection using [Rate Limiting Nullifiers (RLN)](https://rfc.vac.dev/spec/32/).
+The `nwaku` client now contains a working demonstration and integration of RLN relay.
+Check out [this tutorial](https://github.com/status-im/nim-waku/blob/ee96705c7fbe4063b780ac43b7edee2f6c4e351b/docs/tutorial/rln-chat2-live-testnet.md) to see the protocol in action using a toy chat application built on `nwaku`.
```
Mention that we'd love to see people join dogfooding of this as part of operator incentive testnet, call to action to join discord here

oskarth commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+
+### Reaching out to operators:
+
+We want to push for operators to run and maintain their own Waku v2 nodes,
```
...are starting to push for

oskarth commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+We are working on [a set of features](https://github.com/vacp2p/research/issues/97) built into Waku
+that will provide the general security properties Waku users may desire
+and do so in a modern and simple way.
+As a first step, we've already made good progress toward [integrating noise handshakes](https://forum.vac.dev/t/noise-handshakes-as-key-exchange-mechanism-for-waku2/130) as a key exchange mechanism in Waku v2.
```
This is useful for application outside of Status that want similar security guarantees

oskarth commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+This will increase the overall security of the network
+and encourage operators to run their own Waku nodes.
+In turn, the sustainability of Vac as an organization will be better guaranteed.
+As such, protocol incentivization was one of the major conclusions in our recent [Vac Sustainability and Business Workshop](https://forum.vac.dev/t/vac-sustainability-and-business-workshop/).
```
replace conclusion with "focus" (makes it seem like it wasn't important before otherwise)

oskarth commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+with blockchain interaction to manage members,
+punish spammers
+and reward spam detectors.
+After this, we want to design monetary incentivization for providers of `store`, `lightpush` and `filter` services.
```
A big challenge here is in terms of doing it in a private fashion, so we can keep similar metadata protection guarantees as the Waku base layer. This ties into our focus on ZK (link to ZK post on forum)

oskarth commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+including methods to increase security such as merging with the Ethereum Discovery v5 network,
+introducing explicit NAT traversal
+and utilizing [topic advertisement](https://github.com/ethereum/devp2p/blob/fa6428ada7385c13551873b2ae6ad2457c228eb8/discv5/discv5-theory.md#topic-advertisement).
+The [Waku v2 Discovery v5 Roadmap](https://forum.vac.dev/t/waku-v2-discv5-roadmap-discussion/121) contains more details.
```
Oh it is mentioned here I see

oskarth reviewed (approved):
Nice one, covers quite a lot!

oskarth commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+---
+layout: post
+name:  "What's new in `nwaku`?"
```
Not 100% set on this btw, so if you prefer it the way it is or some other title that's fine too

jm-clius replied:
It makes sense to me to call this "Introducing nwaku". Done in [05b607c](https://github.com/vacp2p/vac.dev/pull/55/commits/05b607c2812a885741e7f51bc0b465aff978a26e)

D4nte commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+but it is now official:
+`nim-waku` has been renamed to `nwaku`.
+Why? Well, we needed a recognizable name for our client that could easily be referred to in everyday conversations
+and `nim-waku` just didn't roll off the tongue.
```
Maybe it is worth mentioning that `nim-waku` repo contains both waku v1 and waku v2 implementations and we wanted a name to designate the new codebase.

D4nte replied:
Oh, is `nwaku` *not* only for waku v2? I thought the repo would be kept `nim-waku` but the waku v2 binary would be renamed to `nwaku`.

oskarth replied:
Agree

Don't have any strong opinions on the waku1 binary etc, I don't think a lot of ppl use these so maybe it can be slightly hidden away?

D4nte commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+we've introduced new protocols,
+tweaked our existing protocols
+and expanded our team capacity.
+We've also shown [in a series of practical experiments](./2021-10-25-waku-v1-vs-waku-v2.md) that Waku v2 does indeed deliver on some of the [theoretical advantages](./2020-07-01-waku-v2-pitch.md) it was designed to have over its predecessor, Waku v1.
```
These links are incorrect, need to use the one present in the `permalink` metadata of the target post.

jm-clius replied:
Thanks! Permalinks all fixed in [05b607c](https://github.com/vacp2p/vac.dev/pull/55/commits/05b607c2812a885741e7f51bc0b465aff978a26e)

D4nte commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+or as an encapsulated service node in other applications.
+This post looks at some recent developments within the Nim client.
+
+## 1. `nim-waku` is now known as `nwaku`
```
The css is quite messed up, the backtick does not render well in titles:
![image](https://user-images.githubusercontent.com/300805/162376253-c4c1a014-c40c-417a-b37b-1b37bd0732f8.png)

D4nte replied:
I think we need to get help even with body text back tick, I don't find the rendering great:

![image](https://user-images.githubusercontent.com/300805/162376377-0b713d62-13f6-4c43-a48c-5c48f5769064.png) I'd welcome a grey with margins like GitHub does: 
![image](https://user-images.githubusercontent.com/300805/162376558-66988845-44fc-497d-870f-33df43e9440c.png)

jemboh replied:
How about this? 

<img width="723" alt="image" src="https://user-images.githubusercontent.com/3829925/162738713-3cc128fb-8e00-4233-913c-c657c84ff308.png">

I think I'll sort out the spacing when there are two _blocks_ above/below each other like that but don't want to spend more time if it's not a crowd pleaser.

jm-clius replied:
I removed the backticks in the title in [05b607c](https://github.com/vacp2p/vac.dev/pull/55/commits/05b607c2812a885741e7f51bc0b465aff978a26e) as it's not strictly necessary. Have kept the backticks in the body text, though.

D4nte reviewed (approved):
LGTM.

3 issues related to the website (and not this post) that we need to somehow fix:

- Looks like some manual change is needed for the post to show in /research-log/
- the `backtick` css does not render well
- The titles anchor are broken, ie, when you click on one it scrolls up to the section but does not quite align (the actual target title ends up behind the top banner)

jm-clius commented on _posts/2022-04-11-nwaku-update.md:
```diff
@@ -0,0 +1,277 @@
+In line with our [long term goals](https://vac.dev/#about),
+we've introduced new protocols,
+tweaked our existing protocols
+and expanded our team capacity.
```
Done in [05b607c](https://github.com/vacp2p/vac.dev/pull/55/commits/05b607c2812a885741e7f51bc0b465aff978a26e)

jm-clius commented:
Thanks for the comments, all. I've addressed comments related to content in [05b607c](https://github.com/vacp2p/vac.dev/pull/55/commits/05b607c2812a885741e7f51bc0b465aff978a26e)

> 3 issues related to the website (and not this post) that we need to somehow fix:

Thanks for this, @D4nte. Have clicked around a bit to see if I can find any easy fixes, but a bit out of my depth.

jm-clius commented:
Based on how the post gets rendered I've had to add a couple of formatting workarounds.

Besides the issues mentioned above, I've also noticed:
1. No rendering of headings after level 3 (#### and beyond)
2. No rendering of **bold** text in headings
3. No rendering of `backtick`s in numbered lists
4. No rendering of any formatting at all in post title
